File: LineMate/client/src/components/players/Player.js
import React from "react";
import { Button } from "reactstrap";
import { useHistory } from "react-router-dom";

export default function Player({ player, handleDelete }) {
    const history = useHistory();

    const playerDetails = () => {
        history.push(`/player/${player.id}`)
    }

    return (
        <>
            <tbody>
                <tr>
                    <th scope="row">
                        {player.jerseyNumber}
                    </th>
                    <td className="justify-content-center" tag="h5">
                        {player.lastName}, {player.firstName}
                    </td>
                    <td>
                        {player.position}
                    </td>
                    <td>
                        {player.team?.name}
                    </td>
                    <td>
                        <Button color="info" onClick={playerDetails}>Details</Button>
                    </td>
                </tr>
            </tbody>
        </>
    )
}

File: LineMate/Models/Player.cs
using System.ComponentModel.DataAnnotations;

namespace LineMate.Models
{
    public class Player
    {
        public int Id { get; set; }
        [Required]
        public string FirstName { get; set; }
        [Required]
        public string LastName { get; set; }
        public string Position { get; set; }
        public int JerseyNumber { get; set; }
        public int TeamId { get; set; }
        public int Line { get; set; }
        public Team Team { get; set; }
    }
}